src/test/java/it/com/atlassian/bitbucket/jenkins/internal/fixture/JenkinsProjectHandler.java

```diff
@@ -121,10 +127,11 @@ public void runWorkflowJobForBranch(WorkflowMultiBranchProject mbp, String branc
         Future<WorkflowRun> startCondition = item.scheduleBuild2(0).getStartCondition();
         WorkflowRun workflowRun = startCondition.get(1, TimeUnit.MINUTES);
 
-        while (!workflowRun.equals(item.getLastSuccessfulBuild())) {
-            System.out.println("Waiting for workflow run to finish");
+        while (!workflowRun.equals(item.getLastCompletedBuild())) {
```

Comment:
When something goes wrong in the run, the test doesn't fail until it times out. By using `getLastCompletedBuild` and checking that it is successful afterwards we're able to fail a bit faster